src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -1074,9 +1089,11 @@ - (void)subscribeAttributeWithEndpointId:(NSUInteger)endpointId
             }
         };
 
-        auto chipEndpointId = static_cast<chip::EndpointId>(endpointId);
-        auto chipClusterId = static_cast<chip::ClusterId>(clusterId);
-        auto chipAttributeId = static_cast<chip::AttributeId>(attributeId);
+        auto chipEndpointId
+            = (endpointId == nil) ? kInvalidEndpointId : static_cast<chip::EndpointId>([endpointId unsignedShortValue]);
+        auto chipClusterId = (clusterId == nil) ? kInvalidClusterId : static_cast<chip::ClusterId>([clusterId unsignedLongValue]);
+        auto chipAttributeId
+            = (attributeId == nil) ? kInvalidAttributeId : static_cast<chip::AttributeId>([attributeId unsignedLongValue]);
```

Comment:
Again, might be better to default-allocated AttributePathParams and then set the non-wildcard members.